flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/BaseSimpleTransformationTranslator.java

```diff
@@ -0,0 +1,118 @@
+
+		final Collection<Integer> transformedIds =
+				translateForBatchInternal(transformation, context);
+		configure(transformation, context);
+		return transformedIds;
+	}
+
+	@Override
+	public Collection<Integer> translateForStreaming(final T transformation, final Context context) {
+		checkNotNull(transformation);
+		checkNotNull(context);
+
+		final Collection<Integer> transformedIds =
+				translateForStreamingInternal(transformation, context);
+		configure(transformation, context);
+		return transformedIds;
+	}
+
+	/**
+	 * Translates a given {@link Transformation} to its runtime implementation for
+	 * {@link org.apache.flink.streaming.api.RuntimeExecutionMode#BATCH BATCH-style} execution.
+	 *
+	 * @param transformation The transformation to be translated.
+	 * @param context The translation context.
+	 * @return The ids of the nodes corresponding to this transformation in the transformation graph.
+	 */
+	protected abstract Collection<Integer> translateForBatchInternal(final T transformation, final Context context);
+
+	/**
+	 * Translates a given {@link Transformation} to its runtime implementation for
+	 * {@link org.apache.flink.streaming.api.RuntimeExecutionMode#STREAMING STREAMING-style} execution.
+	 *
+	 * @param transformation The transformation to be translated.
+	 * @param context The translation context.
+	 * @return The ids of the nodes corresponding to this transformation in the transformation graph.
+	 */
+	protected abstract Collection<Integer> translateForStreamingInternal(final T transformation, final Context context);
+
+	private void configure(final T transformation, final Context context) {
+		final StreamGraph streamGraph = context.getStreamGraph();
+		final int transformationId = transformation.getId();
+
+		if (transformation.getBufferTimeout() >= 0) {
+			streamGraph.setBufferTimeout(transformationId, transformation.getBufferTimeout());
+		} else {
+			streamGraph.setBufferTimeout(transformationId, context.getDefaultBufferTimeout());
+		}
+
+		if (transformation.getUid() != null) {
+			streamGraph.setTransformationUID(transformationId, transformation.getUid());
+		}
+		if (transformation.getUserProvidedNodeHash() != null) {
+			streamGraph.setTransformationUserHash(transformationId, transformation.getUserProvidedNodeHash());
+		}
+
+		if (!streamGraph.getExecutionConfig().hasAutoGeneratedUIDsEnabled()) {
+			if (transformation instanceof PhysicalTransformation
+					&& transformation.getUserProvidedNodeHash() == null
+					&& transformation.getUid() == null) {
+				throw new IllegalStateException("Auto generated UIDs have been disabled " +
+						"but no UID or hash has been assigned to operator " + transformation.getName());
+			}
+		}
+
+		if (transformation.getMinResources() != null && transformation.getPreferredResources() != null) {
+			streamGraph.setResources(transformationId, transformation.getMinResources(), transformation.getPreferredResources());
+		}
+
+		streamGraph.setManagedMemoryUseCaseWeights(
```

Comment:
Just a quick comment here. I think we might need to set different weights/use cases depending on the runtime mode. E.g. in the BATCH runtime we will declare `ManagedMemoryUseCase.BATCH_OP`, whereas for streaming `ManagedMemoryUseCase.ROCKS_DB`.

I am still not entirely sure, how this should all play together, but thats a concern that I have.